### ISSUE TITLE
Update axis in drillup event

### ISSUE THREAD

**Issue by pawelfus** [highslide-software/highcharts.com#3733]
Example: http://jsfiddle.net/y47xer7b/8/
Steps:

- click on food
- go back to first level

Now, points' names aren't treated as categories on xAxis. When removing `xAxis.setTitle()` method from `drillup` callback, it works properly. 

Workaround may be to use separate axis for each series: http://jsfiddle.net/y47xer7b/10/

**Comment by stephanevanraes** [comment]
Simplified: http://jsfiddle.net/stephanevanraes/ut22aqth/
It is not related to the drilldown module

**Comment by stephanevanraes** [comment]
Since there are no categories defined, Highcharts pulls the category names from the data itself.  This is done in **setData** on the series and stored in the *names array*.  Updating the title causes highcharts to destroy and recreate the axis, since the series already exist **setData** is not called and the names array remains empty. 

A work around would be save the names array and restore it into the axis.
On a side note, each setTitle causes a redraw of the chart so you should called it with redraw false and at the end redraw the chart.

The workaround can be found here: http://jsfiddle.net/stephanevanraes/y47xer7b/11/ or simplified http://jsfiddle.net/stephanevanraes/ut22aqth/2/

**Comment by pawelfus** [comment]
Yes, of course, updating should be done without redraw. However that doesn't affect the issue.

Even simpler workaround would be just to call `series.update()`, to force redrawing axis names: http://jsfiddle.net/ut22aqth/3/ (won't work for drilldown demo, though).

**Comment by highslide-software** [comment]
A possible solution would be to refactor the whole `names` logic, and provide the Axis object itself with a new method, `getNames`. This would be called after all data updates, like Series.setData, Point.update, Series.addPoint etc (listen to the `udatedData` hook?). It would simply loop over all the axis.series and look for point names. 

But in that case we need to consider #2522 too.
